### PR TITLE
Prediction: extract trajectory from semantic_LSTM

### PR DESCRIPTION
**Pull request by HongyiSun**

### CODE REVIEW

File: modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc [hunk 1]
```diff
@@ -30,17 +30,131 @@
...
+    ADEBUG << "Obstacle [" << id << "] failed to extract obstacle history";
+    return false;
+  }
+  // Process obstacle_history
+  torch::Tensor obstacle_pos = torch::zeros({1, 20, 2});
+  torch::Tensor obstacle_pos_step = torch::zeros({1, 20, 2});
+  for (int i = 0; i < 20; ++i) {
+    obstacle_pos[0][19 - i][0] = pos_history[i].first;
+    obstacle_pos[0][19 - i][1] = pos_history[i].second;
+    if (i == 19 || (i > 0 && pos_history[i].first == 0.0)) break;
```
Comment (kechxu) [inline modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc]: coding style

File: modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc [hunk 2]
```diff
@@ -30,17 +29,132 @@
...
+
+  Clear();
+  CHECK_NOTNULL(obstacle_ptr);
+  int id = obstacle_ptr->id();
+  if (!obstacle_ptr->latest_feature().IsInitialized()) {
+    AERROR << "Obstacle [" << id << "] has no latest feature.";
+    return false;
+  }
+  Feature* latest_feature_ptr = obstacle_ptr->mutable_latest_feature();
+  CHECK_NOTNULL(latest_feature_ptr);
```
Comment (xiaoxq) [inline modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc]: No need.

File: modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc [hunk 3]
```diff
@@ -30,17 +29,132 @@
...
-bool SemanticLSTMEvaluator::Evaluate(Obstacle* obstacle_ptr) { return true; }
+bool SemanticLSTMEvaluator::Evaluate(Obstacle* obstacle_ptr) {
+  omp_set_num_threads(1);
+
+  obstacle_ptr->SetEvaluatorType(evaluator_type_);
+
+  Clear();
+  CHECK_NOTNULL(obstacle_ptr);
+  int id = obstacle_ptr->id();
+  if (!obstacle_ptr->latest_feature().IsInitialized()) {
```
Comment (xiaoxq) [inline modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc]: Why not `has_latest_feature()`?

File: modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc [hunk 4]
```diff
@@ -30,17 +29,132 @@
...
+    obstacle_pos[0][19 - i][0] = pos_history[i].first;
+    obstacle_pos[0][19 - i][1] = pos_history[i].second;
+    if (i == 19 || (i > 0 && pos_history[i].first == 0.0)) {
+      break;
+    }
+    obstacle_pos_step[0][19 - i][0] =
+        pos_history[i].first - pos_history[i + 1].first;
+    obstacle_pos_step[0][19 - i][1] =
+        pos_history[i].second - pos_history[i + 1].second;
+  }
```
Comment (xiaoxq) [inline modules/prediction/evaluator/vehicle/semantic_lstm_evaluator.cc]: Avoid magic numbers. Give `20` a name such as `N`, then use `N - 1` to replace `19`.

**Pull request merged**